tests/integration/network.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/api/jsonrpc"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+var (
+	ErrUnableToConfirmTx = errors.New("unable to confirm transaction")
+	ErrInvalidURI        = errors.New("invalid uri")
+	ErrTxNotFound        = errors.New("tx not found")
+)
+
+const (
+	txCheckInterval = 100 * time.Millisecond
+)
+
+type Network struct {
+	uris []string
+}
+
+func (*Network) ConfirmTxs(ctx context.Context, txs []*chain.Transaction) error {
+	return instances[0].ConfirmTxs(ctx, txs)
+}
+
+func (*Network) GenerateTx(ctx context.Context, actions []chain.Action, auth chain.AuthFactory) (*chain.Transaction, error) {
+	return instances[0].GenerateTx(ctx, actions, auth)
+}
+
+func (i *instance) ConfirmTxs(ctx context.Context, txs []*chain.Transaction) error {
+	errs := i.vm.Submit(ctx, true, txs)
+	if len(errs) != 0 && errs[0] != nil {
+		return errs[0]
+	}
+
+	expectBlk(i)(false)
+
+	for {
+		allFound := true
+		for _, tx := range txs {
+			err := i.confirmTx(ctx, tx.ID())
+			if err == ErrTxNotFound {
+				allFound = false
+				break
+			}
+			return err
+		}
+		if allFound {
+			return nil
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.NewTimer(txCheckInterval).C:
+			// try again
+		}
+	}
```

Comment:
There's no need for retries here. Block building/acceptance will only be triggered by `expectBlk`. If something goes wrong where the tx is not included, it will never be included.

Reply:
done